# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382161206 25200
#      Fri Oct 18 22:40:06 2013 -0700
# Node ID 1efe4c65c5cffad730dcbba76ad945e758f9b479
# Parent  efbe746f8fa89a6870f6b24ba89b918addd34c6d
Deprecate interp1q.m.

* scripts/deprecated/interp1q.m: Moved from scripts/general.

* scripts/deprecated/module.mk, scripts/general/module.mk: Update build system

* NEWS: Announce deprecation.

* doc/interpreter/interp.txi: Remove interp1q from manual.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -317,24 +317,25 @@ Summary of important user-visible change
       betai      gammai                perror       values
       cellidx    glpkmex               replot               
       cquad      is_duplicate_entry    saveimage            
       
     The following functions have been deprecated in Octave 3.8 and will
     be removed from Octave 3.12 (or whatever version is the second major
     release after 3.8):
 
-      default_save_options    java_set                  
-      gen_doc_cache           java_unsigned_conversion  
+      default_save_options    java_new            
+      gen_doc_cache           java_set                   
+      interp1q                java_unsigned_conversion
       isequalwithequalnans    javafields                
       java_convert_matrix     javamethods               
       java_debug              re_read_readline_init_file
       java_get                read_readline_init_file   
       java_invoke             saving_history            
-      java_new           
+
 
     The following keywords have been deprecated in Octave 3.8 and will
     be removed from Octave 3.12 (or whatever version is the second major
     release after 3.8):
 
       static
 
     The following configuration variables have been deprecated in Octave
diff --git a/doc/interpreter/interp.txi b/doc/interpreter/interp.txi
--- a/doc/interpreter/interp.txi
+++ b/doc/interpreter/interp.txi
@@ -77,22 +77,16 @@ step function}
 
 @float Figure,fig:interpderiv2
 @center @image{interpderiv2,4in}
 @caption{Comparison of the second derivative of the @qcode{"pchip"} and @qcode{"spline"} 
 interpolation methods for a step function}
 @end float
 @end ifnotinfo
 
-A simplified version of @code{interp1} that performs only linear
-interpolation is available in @code{interp1q}.  This argument is slightly
-faster than @code{interp1} as to performs little error checking.
-
-@DOCSTRING(interp1q)
-
 Fourier interpolation, is a resampling technique where a signal is
 converted to the frequency domain, padded with zeros and then
 reconverted to the time domain.
 
 @DOCSTRING(interpft)
 
 There are two significant limitations on Fourier interpolation.  Firstly,
 the function signal is assumed to be periodic, and so non-periodic
diff --git a/scripts/general/interp1q.m b/scripts/deprecated/interp1q.m
rename from scripts/general/interp1q.m
rename to scripts/deprecated/interp1q.m
--- a/scripts/general/interp1q.m
+++ b/scripts/deprecated/interp1q.m
@@ -32,16 +32,24 @@
 ## Note that the error checking is only a significant portion of the
 ## execution time of this @code{interp1} if the size of the input arguments
 ## is relatively small.  Therefore, the benefit of using @code{interp1q}
 ## is relatively small.
 ## @seealso{interp1}
 ## @end deftypefn
 
 function yi = interp1q (x, y, xi)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "interp1q is obsolete and will be removed from a future version of Octave; use interp1 instead");
+  endif
+
   x = x(:);
   nx = rows (x);
   szy = size (y);
   y = y(:,:);
   [ny, nc] = size (y);
   szx = size (xi);
   xi = xi (:);
   dy = diff (y);
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -1,22 +1,23 @@
 FCN_FILE_DIRS += deprecated
 
 deprecated_FCN_FILES = \
   deprecated/__error_text__.m \
   deprecated/cor.m \
   deprecated/corrcoef.m \
   deprecated/cut.m \
   deprecated/default_save_options.m \
-  deprecated/java_debug.m \
   deprecated/error_text.m \
   deprecated/gen_doc_cache.m \
+  deprecated/interp1q.m \
   deprecated/isequalwithequalnans.m \
   deprecated/isstr.m \
   deprecated/java_convert_matrix.m \
+  deprecated/java_debug.m \
   deprecated/java_get.m \
   deprecated/java_invoke.m \
   deprecated/java_new.m \
   deprecated/java_unsigned_conversion.m \
   deprecated/java_set.m \
   deprecated/javafields.m \
   deprecated/javamethods.m \
   deprecated/polyderiv.m \
diff --git a/scripts/general/module.mk b/scripts/general/module.mk
--- a/scripts/general/module.mk
+++ b/scripts/general/module.mk
@@ -32,17 +32,16 @@ general_FCN_FILES = \
   general/flipdim.m \
   general/fliplr.m \
   general/flipud.m \
   general/genvarname.m \
   general/gradient.m \
   general/idivide.m \
   general/int2str.m \
   general/interp1.m \
-  general/interp1q.m \
   general/interp2.m \
   general/interp3.m \
   general/interpn.m \
   general/interpft.m \
   general/isa.m \
   general/iscolumn.m \
   general/isdir.m \
   general/isequal.m \
